modules/apigee_edge_apidocs/src/ApiDocListBuilder.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+/**
+ * Copyright 2018 Google Inc.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * version 2 as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,
+ * MA 02110-1301, USA.
+ */
+
+namespace Drupal\apigee_edge_apidocs;
+
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\Core\Entity\EntityTypeInterface;
+use Drupal\Core\Link;
+use Drupal\Core\Entity\EntityListBuilder;
+use Drupal\Core\Url;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Defines a class to build a listing of API Doc entities.
+ *
+ * @ingroup apigee_edge_apidocs
+ */
+class ApiDocListBuilder extends EntityListBuilder {
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function createInstance(ContainerInterface $container, EntityTypeInterface $entity_type) {
+    return new static(
+      $entity_type,
+      $container->get('entity.manager')->getStorage($entity_type->id())
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   *
+   * We override ::render() so that we can add our own content above the table.
+   * parent::render() is where EntityListBuilder creates the table using our
+   * buildHeader() and buildRow() implementations.
+   */
+  public function render() {
+    $build['description'] = [
+      '#markup' => $this->t('Manage your API documentation. You can manage the fields on the <a href=":url">API Docs settings page</a>.', [
+        ':url' => Url::fromRoute('apidoc.settings')->toString(),
+      ]),
+    ];
+    $build['table'] = parent::render();
+    return $build;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildHeader() {
+    $header['id'] = $this->t('ID');
+    $header['name'] = $this->t('Name');
```

Comment:
Would it be worth exposing the description as additional column?

Reply:
I created an extra ticket for that: https://apigeesc.atlassian.net/browse/DRUP-618